Jarvis-Platform/jarvis-studio
# Feature/notes thread

## Description
Add the thread functionality on a note.


## Todos
- [x] Reorganize components structure
- [x] Add a note on a thread
- [x] Modify a note on a thread
- [x] Delete a note on a thread
- [ ] Delete parent note behavior => Delete all children with CF watching for snapshot changes (this will provide us a way to send notifications to users)


## Deploy Notes
Review & merge


## Steps to Test or Reproduce
```sh
git checkout master
git pull
git checkout feature/notes-thread
```

1. Go to any `ItemView` that as notes or create note(s) on an item
2. Start a thread
3. Insert a note
4. Try to edit the note
5. Delete parent note => should be marked as deleted
6. Delete all notes in thread => parent note should be deleted

## Impacted Areas in Application
- All `ItemView`
- `src/store/modules/easy-firestore`
- `src/components/data-workflows/common/item/notes`
